### DISCUSSION TITLE
Naming CI workflows and jobs

### DISCUSSION THREAD

**Discussion by szepeviktor** [diesel-rs/diesel#3822] [Ideas]
Hello!

Here is how CI UI looks like currently.
![image](https://github.com/diesel-rs/diesel/assets/952007/3854d93a-fd50-416e-a641-cf5f6ed5f2dd)

Excuse me but these names look like expletives without real meaning.
I'd like to improve these names.

Name workflows with a noun and jobs with a verb. For example "Documentation / Publish".

This is from my https://github.com/szepeviktor/byte-level-care repository.
![image](https://github.com/diesel-rs/diesel/assets/952007/8f12200f-0967-4b2f-b525-5e0cd118f555)

What kind of tests are "CI Tests / Check"? Unit? Functional? End-to-end? Mix of these?

**Comment by weiznich** [comment]
We are happy to accept a PR that improves this naming.

**Reply by szepeviktor** [reply]
Could you help me answer my question?

**Reply by weiznich** [reply]
I think you refer to this as a question:

> What kind of tests are "CI Tests / Check"? Unit? Functional? End-to-end? Mix of these?

In that case the answer is: The names don't stop with "CI Tests/Check", but rather continue afterwards. As shown in your image there are the following cases:

* "CI Tests / Check rustfmt style && run clippy" which runs `cargo fmt --check` and `cargo clippy` on the project
* "CI Tests / Check sqlite bundled + Sqlite with asan" which checks whether or not the sqlite bundled configuration builds and runs most of the tests there using memory sanitizers (address sanitizer -> asan).

**Reply by szepeviktor** [reply]
> and runs most of the tests there

Yes. I am interested in these tests.
Are these unit, functional or end-to-end tests?

**Reply by szepeviktor** [reply]
I suspect these are functional test because they include a database back-end (not unit tests) but does not use diesel as in real life (not e2e tests).

**Reply by weiznich** [reply]
These tests include unit test, integration tests and end 2 end tests. There are several steps involved in this job, which test different things each time.
